nixos/mailman: add http listener for uwsgi

petabyteboy opened the pull request:
With the config suggested in the module docs both Mailman core and
Hyperkitty are running, but Mailman core can not connect to Hyperkitty,
since the default hyperkitty.baseUrl is not set up by the module.

This adds a http listener to the uwsgi config and changes the default
hyperkitty.baseUrl to connect to this http listener.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

alyssais commented:
> With the config suggested in the module docs both Mailman core and
> Hyperkitty are running, but Mailman core can not connect to Hyperkitty,
> since the default hyperkitty.baseUrl is not set up by the module.
>
> This adds an http proxy for the default baseUrl
> http://localhost/archiver/, so the default config can have working mail

Typo in your commit message here.  Should say "/archives/".

> archiving.

Would it make more sense to set this up in uwsgi directly, rather than
nginx, so we're not going through two layers of HTTP servers?

In my configuration (which doesn't use services.mailman.service yet), I
do it like this:

    services.mailman.hyperkitty.baseUrl = "http://localhost:18507/lists/hyperkitty";
    services.uwsgi.instance = {
      http = "localhost:18507";
      # ...
    };

petabyteboy commented:
Okay, sure, if we can do it in uwsgi directly that would be even better! I will try to build that, but I have no idea about uwsgi and I think the services.mailman module doesn't use services.uwsgi.

alyssais commented:
> Okay, sure, if we can do it in uwsgi directly that would be even
> better! I will try to build that, but I have no idea about uwsgi and I
> think the services.mailman module doesn't use services.uwsgi.

services.uwsgi is a very (sometimes unpleasantly) thin wrapper around
the uWSGI config file, so it should be very similar. :)

Commit:
nixos/mailman: add http listener for uwsgi

With the config suggested in the module docs both Mailman core and
Hyperkitty are running, but Mailman core can not connect to Hyperkitty,
since the default hyperkitty.baseUrl is not set up by the module.

This adds a http listener to the uwsgi config and changes the default
hyperkitty.baseUrl to connect to this http listener.

petabyteboy commented:
I changed it to use the builtin uwsgi http listener, it wasn't too hard :)
Please re-review.

petabyteboy commented:
Also should we get rid of the socket now and use the http listener for everything, or keep both?

alyssais commented:
Difficult for me to test this with my setup since I'm not using the
serve option, but the diff LGTM so I'm happy for you to merge if it
works for you.

> Also should we get rid of the socket now and use the http listener for
> everything, or keep both?

I think we should keep both and use the socket where possible for
simplicity and (presumably) reduced overhead.
